tools/azure-devtools/src/azure_devtools/scenario_tests/recording_processors.py

```diff
@@ -238,6 +229,8 @@ def process_response(self, response):
             self.replace_header(response, 'azure-asyncoperation', old, new)
             self.replace_header(response, "www-authenticate", old, new)
 
+        response["url"] = replace_subscription_id(response["url"])
```

Comment:
SubscriptionRecordingProcessor already does this, do we need to do it again here?

Reply:
`SubscriptionRecordingProcessor` is not used by default in the `AzureMgmtTestCase` class. It probably should be but I'd rather do that in a separate PR.

Reply:
#18981